src/gc/env/gcenv.base.h

```diff
@@ -281,6 +283,58 @@ inline uint8_t BitScanForward64(uint32_t *bitIndex, uint64_t mask)
 #endif // _MSC_VER
 }
 
+// Cross-platform wrapper for the _BitScanReverse compiler intrinsic.
+inline uint8_t BitScanReverse(uint32_t *bitIndex, uint32_t mask)
```

Comment:
In gcenv.base.h `BitScanReverse` functions are declared to return `unsigned char`, in pal.h — `uint8_t`, while they actually return `TRUE`/`FALSE`.  That is confusing.

Reply:
I agree. The existing ``BitScanForward`` functions are defined in the same way. Ideally, they'd all be changed to return ``bool`` -- is that a change you want me to make as part of this PR, or should we circle back and make that as a separate change?

Reply:
This is how the Windows version of _BitScanReverse / Forward are defined. They return unsigned char, which for Unix we express as uint8_t. I am not sure if it is worth diverging from the Windows signature. We could change the return type to "unsigned char", but then it would look weird to have parameters specified by the explicit size types and the return to be different.

Reply:
Resolving per @janvorli's comment.